.github/workflows/check_amalgamation.yml

```diff
@@ -0,0 +1,92 @@
+name: "Check amalgamation"
+
+on:
+  pull_request_target:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    env:
+      MAIN_DIR: ${{ github.workspace }}/main
+      INCLUDE_DIR: ${{ github.workspace }}/main/single_include/nlohmann
+      TOOL_DIR: ${{ github.workspace }}/tools/tools/amalgamate
+      ASTYLE_FLAGS: >
```

Comment:
This reminds me that I'd really want to switch to Clang-Format at some point...

Reply:
Yes, I'm ready to go, just waiting for the custom base class and serialization macro PRs to be merged.